### ISSUE TITLE
wildcard or startsWith possibility in filter_by?

### ISSUE THREAD

**Issue by s-gal** [typesense/typesense#1092]
As far as i understand the Documentation you can build the filter_by like this for matching:
`country: USA`

and for exact matching like this:
`category:= Shoe`

We face an issue that we need to build a scoped key to limit users to only see their data. let's name the field customer with these values:

1, "BMW"
2, "BMW Retail Group"
3, "Retail Group Mercedes Germany"
4, "Retail Group Mercedes Austria"

One Customers should be allowed to see all data for "Retail Group". Which would be entry 3 and 4.

If i define the filter though with 
`customer: Retail Group`
it will also match entry 2.

`customer:= Retail Group`
would match none of the above....

So it would be great if the be a wildcard- or a starstWith-parameter to make this filter possible:
`customer:= Retail Group*`

Is there maybe already something like that, that i can apply and i oversee in the documentation of typesense?

If not this would be a strongly desired feature request.

**Comment by jasonbosco** [comment]
This is not possible to do in Typesense at the moment, but I'll leave this open as a feature request. 

One workaround in the meantime would be to see if you're able to parse words like "Retail Group", etc that you would want to filter on, into a new field at indexing time and then filter on that structured field.

**Comment by leerobert** [comment]
Or just use an "identifier" for each of these. Map BMW retail group -> 1 BMW -> 2, etc. Maintain a secondary lookup table.

**Comment by s-gal** [comment]
Thanks for the input and your suggestions. I had a workaround as @jasonbosco mentioned also in mind, hoped that i missed something in the documentation though and there would bee already a wildcard- or startsWith-keyword. ChatGPT also trolled me and said there is already something like that :)

Yes, would be great to see such a possibility in a future version.

**Comment by rajaie** [comment]
I believe this can now be done using prefix filtering https://typesense.org/docs/27.1/api/search.html#filter-parameters

**Comment by jasonbosco** [comment]
That's right! Thank you for hunting this issue down and posting an update @rajaie!

Duplicate of #696
